### PR TITLE
import: improve dividends matching

### PR DESCRIPTION
**Pull request by iliakan**
Fixes CHOICE DIVIDEND matching for withheld tax.

### CODE REVIEW

File: jal/data_import/statement_ibkr.py [hunk 1]
```diff
@@ -661,28 +662,28 @@ def loadIBDepositWithdraw(self, cash):
...
 
-    def findDividend4Tax(self, timestamp, account_id, asset_id, note):
+    def findDividend4Tax(self, timestamp, account_id, asset_id, symbol, isin, amount):
         # Check strong match
         id = readSQL("SELECT id FROM dividends WHERE type=:div AND timestamp=:timestamp "
-                     "AND account_id=:account_id AND asset_id=:asset_id AND note LIKE :dividend_description",
+                     "AND account_id=:account_id AND asset_id=:asset_id AND regexp(:dividend_regexp, note)",
                      [(":div", DividendSubtype.Dividend), (":timestamp", timestamp), (":account_id", account_id),
-                      (":asset_id", asset_id), (":dividend_description", note)])
+                      (":asset_id", asset_id), (":dividend_regexp", rf"^{symbol}\s*\({isin}\).*?DIVIDEND.*?{amount}")])
```
Comment (iliakan) [inline jal/data_import/statement_ibkr.py:686]: Maybe add a test: there must he at most one matching id

**Comment by titov-vv** [comment]
Big thanks for PR - it is very good indeed but tests were not passed unfortunately. The problem happens when tax/div description has no amount within (e.g. like `"XXX(ISIN) PAYMENT IN LIEU OF DIVIDEND - US TAX"`).
I think I'll change `findDividend4Tax()` method in one of 2 ways:
- either I'll keep 'strong match' branch (it was because it is proven to work well), but I'll enhance 'weak match' branch with regex as you proposed. 
- or I'll keep only 'weak match' branch with a bit more complex regex.  

Anyway, your proposal of `REGEX` usage instead of `LIKE` is correct and should be applied everywhere in code.

**Comment by iliakan** [comment]
The PR was updated a bit. I also created an issue about payments in lieu.

How to run tests? I missed them very much, it's so good you have them!

**Comment by titov-vv** [comment]
Tests are still in my todo list :( 
I have a set of XML files (I can't share it as they are real with private data). I'll find some time to combine one XML from them with all specific cases and then if will be possible to make an automatic test for it. Will try to do it soon but it will require some manual work...

**Comment by titov-vv** [comment]
I did some test and merget as it looks ok now.
Next step is to extend automatic test set to have these things validated automatically. It will take some time so I decided to merge after manual testing.

**Comment by iliakan** [comment]
Please take a look at the data I sent you, to ensure correct dividends.

**Comment by titov-vv** [comment]
Sure. More validation to come.

**Pull request merged**